src/test/java/hudson/plugins/deploy/PipelineSyntaxTest.java

```diff
@@ -0,0 +1,133 @@
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.steps.CoreStep;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.util.Collections;
+
+/**
+ * Tests pipeline compatibility. Since there ia no Builder that sets the build status all of these tests
+ * will ultimately result in a no-op.
+ */
+public class PipelineSyntaxTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    private String getFullScript (String func) {
+        return "node {\n" +
+                    "writeFile(file: 'readme.txt', text: 'this creates a workspace if one doesnt already exist')\n" +
+                    func +
+                "}";
+    }
+
+    @Test
+    public void testNoAdapterDeploy() throws Exception {
+        WorkflowJob p = j.getInstance().createProject(WorkflowJob.class, "DryRunTest");
+        p.setDefinition(new CpsFlowDefinition(
+                getFullScript("deploy(war: 'target/app.war', contextPath: 'app', onFailure: false)"),
+                false));
+        WorkflowRun b = j.assertBuildStatusSuccess(p.scheduleBuild2(0));
+    }
+
+    @Test
+    public void testMockAdapterDeploy() throws Exception {
+        WorkflowJob p = j.getInstance().createProject(WorkflowJob.class, "MockTest");
+        p.setDefinition(new CpsFlowDefinition(
+                getFullScript("deploy(adapters: [workflowAdapter()], war: 'target/app.war', contextPath: 'app')"),
+                false));
+        WorkflowRun b = j.assertBuildStatusSuccess(p.scheduleBuild2(0));
+    }
+
+    @Test
+    public void testMockAdaptersDeploy() throws Exception {
+        WorkflowJob p = j.getInstance().createProject(WorkflowJob.class, "MockTest");
+        p.setDefinition(new CpsFlowDefinition(
+                getFullScript("deploy(adapters: [workflowAdapter(), workflowAdapter(), workflowAdapter()], war: 'target/app.war', contextPath: 'app')"),
+                false));
+        WorkflowRun b = j.assertBuildStatusSuccess(p.scheduleBuild2(0));
+    }
+
+    @Test
+    public void testGlassFishAdapter() throws Exception {
+        WorkflowJob p = j.getInstance().createProject(WorkflowJob.class, "GlassfishTest");
+        p.setDefinition(new CpsFlowDefinition(
+                getFullScript(
+                "def gf2 = glassfish2( " +
+                    "home: 'FAKE', " +
+                    "credentialsId: 'FAKE'," +
+                    "adminPort: '1234', " +
+                    "hostname: 'localhost') \n" +
+                "def gf3 = glassfish3( " +
+                    "home: 'FAKE', " +
+                    "credentialsId: 'FAKE'," +
+                    "adminPort: '1234', " +
+                    "hostname: 'localhost') \n" +
+                "deploy(adapters: [gf2, gf3], war: 'target/app.war', contextPath: 'app')"),
+                false));
+
+        WorkflowRun b = j.assertBuildStatusSuccess(p.scheduleBuild2(0));
```

Comment:
I named these tests `PipelineSyntaxTest` since they only test the syntax and not the functionality. Yes, this would log a warning but not fail build.